packages/maskbook/src/extension/options-page/DashboardComponents/WalletAssetsTable.tsx

```diff
@@ -336,12 +336,19 @@ export function WalletAssetsTable(props: WalletAssetsTableProps) {
                     </TableBody>
                 </Table>
             </TableContainer>
-            <LessButton
-                isMobile={isMobile}
-                setViewLength={setViewLength}
-                setPrice={setPrice}
-                detailedTokens={detailedTokens}
-            />
+            {detailedTokens.filter((x) =>
```

Comment:
The token filtering logical was using twice. Please refactor it to only use once. A easy catch up solution is like:

```typescript
const visibleTokens = detailedTokens.filter(...)

// Then you can use visibleTokens directly, no need to filter detailedToken again.
```